cmake/modules/configuration_files.cmake

```diff
@@ -78,9 +78,15 @@ else()
   message(FATAL_ERROR "No prj.conf file was found in the ${APPLICATION_CONFIG_DIR} folder, "
                       "please read the Zephyr documentation on application development.")
 endif()
+zephyr_file_suffix(CONF_FILE)
 
 if(CONF_FILE_INCLUDE_FRAGMENTS)
-  zephyr_file(CONF_FILES ${APPLICATION_CONFIG_DIR}/boards KCONF CONF_FILE BUILD ${CONF_FILE_BUILD_TYPE})
+  # FILE_SUFFIX will take priority over a prj_<X> build type if both are used
+  if(DEFINED FILE_SUFFIX)
+    zephyr_file(CONF_FILES ${APPLICATION_CONFIG_DIR}/boards KCONF CONF_FILE BUILD ${FILE_SUFFIX})
```

Comment:
Passing `FILE_SUFFIX` as a build type looks hacky.

Reply:
I didn't want to add another variable to the function just to duplicate the code when it would be doing the same thing. Could rename it perhaps?